server.go

```diff
@@ -0,0 +1,260 @@
+	r.Use(middleware.Recoverer)
+
+	if s.EnableCORS {
+		Logger.Infoln("Enabling CORS support")
+		cors := cors.New(cors.Options{
+			AllowedOrigins:   []string{"*"},
+			AllowedMethods:   []string{"GET", "OPTIONS"},
+			AllowedHeaders:   []string{"Accept", "Accept-Encoding", "Authorization"},
+			AllowCredentials: true,
+		})
+		r.Use(cors.Handler)
+	}
+
+	//-- ROUTES
+	r.Get("/{layers}/{z}/{x}/{y}.mvt", s.cached(s.getVectorTile))
+
+	// TODO: Add GeoJSON endpoint?
+
+	i := chi.NewRouter()
+	i.Get("/healthcheck", s.healthCheck)
+
+	go func() {
+		log.Fatalln(http.ListenAndServe(fmt.Sprintf(":%d", s.Port), r))
+	}()
+
+	go func() {
+		log.Fatalln(http.ListenAndServe(fmt.Sprintf(":%d", s.InternalPort), i))
+	}()
+
+	Logger.Infof("Tilenol server up and running @ 0.0.0.0:[%d,%d]", s.Port, s.InternalPort)
+
+	select {}
+}
+
+// healthCheck implements a simple healthcheck endpoint for the internal metrics server
+func (s *Server) healthCheck(w http.ResponseWriter, r *http.Request) {
+	// TODO: Maybe in the future check that ES is reachable?
+	fmt.Fprintf(w, "OK")
+}
+
+// calculateSimplificationThreshold determines the simplification threshold based on the
+// current zoom level
+func calculateSimplificationThreshold(minZoom, maxZoom, currentZoom int) float64 {
+	s := MinSimplify - MaxSimplify
+	z := float64(maxZoom - minZoom)
+	p := s / z
+	return p*float64(currentZoom-minZoom) + MaxSimplify
+}
+
+// cached is a wrapper function that optionally tries to cache outgoing responses from
+// a Handler
+func (s *Server) cached(handler Handler) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		ctx := r.Context()
+		defer func() {
+			if ctx.Err() == context.Canceled {
+				Logger.Debugf("Request canceled by client")
+				w.WriteHeader(499)
+				return
+			}
+		}()
+
+		// Set standard response headers
+		// TODO: Use the cache TTL to determine the Cache-Control
+		w.Header().Set("Cache-Control", "max-age=86400")
+		w.Header().Set("Content-Encoding", "gzip")
+		// TODO: Store the content type somehow in the cache?
+		w.Header().Set("Content-Type", "application/x-protobuf")
+
+		if s.Cache != nil {
+			key := r.URL.RequestURI()
+			if s.Cache.Exists(key) {
+				Logger.Debugf("Key [%s] found in cache", key)
+				val, err := s.Cache.Get(key)
+				if err != nil {
+					s.handleError(err.(error), w, r)
+					return
+				}
+				buffer := bytes.NewBuffer(val)
+				io.Copy(w, buffer)
+			} else {
+				Logger.Debugf("Key [%s] is not cached", key)
+				var buffer bytes.Buffer
+				herr := handler(ctx, &buffer, r)
+				if herr != nil {
+					return
+				}
+				err := s.Cache.Put(key, buffer.Bytes())
+				if err != nil {
+					// Log an error in case the key can't be stored in cache, but continue
+					Logger.Errorf("Could not store key [%s] in cache: %v", key, err)
+				}
+				_, err = io.Copy(w, &buffer)
+			}
+		} else {
+			handler(ctx, w, r)
+		}
+	}
+}
+
+// filterLayersByNames filters the tile server layers by the names of layers being
```

Comment:
It's interesting to me that this is the de facto way of writing Go docstrings, at least according to [this](https://blog.golang.org/godoc-documenting-go-code)

Reply:
Yeah, it's checked by the linter. It's kind of nice as it makes things consistent, but also can get into your head around how you name functions, knowing that the docstring has to start with its name.